src/Yaapii.Atoms/Map/Sorted.cs

```diff
@@ -0,0 +1,178 @@
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<Key, Value>> pairs)
+            : this(pairs, Comparer<Key>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<Key, Value>> pairs, Func<KeyValuePair<Key, Value>, KeyValuePair<Key, Value>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<Key, Value>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<Key, Value>> pairs, IComparer<Key> cmp)
+            : this(pairs, new KeyComparer<Key, Value>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<Key, Value>> pairs, IComparer<KeyValuePair<Key, Value>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<Key, Value>>(pairs);
+                    items.Sort(cmp);
+                    return new Map.Of<Key, Value>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    public sealed class Sorted<Value> : Map.Envelope<Value>
+    {
+        /// <summary>
+        /// Sorts the given map with the default comperator of the key
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs)
+            : this(pairs, Comparer<string>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, Func<KeyValuePair<string, Value>, KeyValuePair<string, Value>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<string, Value>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<string> cmp)
```

Comment:
we should avoid these ctors accepting IEnumerable of KeyValuePairs because they often confuse the developers.
IDictionary implements IEnumerable. This means this ctor will enumerate all values. This is by default rejected by LazyDict which is the core map for our maps.

So either remove this ctor OR add another ctor which accepts a IDictionary, this will have higher weight than this one and then you can avoid enumerating all values.